raiden/network/rpc/smartcontract_proxy.py

```diff
@@ -119,7 +119,6 @@ def transact(
                         " already pending. Could not find the transaction in the "
                         "local transaction pool. Bailing ..."
                     )
-                return txhash
```

Comment:
Nice find. But I have a question/request.

Now, as you can also see in your test this will end up throwing a `ValueError`. I would like us to generally avoid generic exceptions such as `ValueError` in the codebase. Can you perhaps add a new exception? For example `EthereumNonceTooLow()` ?

Reply:
Do you know where is the documentation for the given error message? I wouldn't assume the error message implies this is always the case.

Reply:
@hackaugusto no and I have had no luck with their documentation. We are talking about `parity` and `"Transaction with the same hash was already imported"` right?

So it's either the transaction is already pending, in which case we already throw the `TransactionAlreadyPending` exception or ... as you just saw the nonce is too low.

I can't answer it with certainty, just say what we know:

It may be:

- Nonce too low
- Some other error that causes `"Transaction with the same hash was already imported"`

If you can think of a good name that encompasses both we can use it ofcourse.

Maybe `ParityTxHashAlreadyImported()` and in the error message ask the user if the given nonce was too low?